addons/account/tests/test_account_move_entry.py

```diff
@@ -485,3 +486,158 @@ def test_misc_prevent_unlink_posted_items(self):
+                    'account_id': tax_waiting_account.id,
+                    'debit': 0.0,
+                    'credit': 100.0,
+                    'tag_ids': [(6, 0, tax_tags['invoice']['tax'].ids)],
+                    'tax_repartition_line_id': tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'tax').id,
+                }),
+                (0, None, {
+                    'name': 'counterpart line',
+                    'account_id': self.company_data['default_account_receivable'].id,
+                    'debit': 1100.0,
+                    'credit': 0.0,
+                }),
+            ]
+        })
+        move.post()
+        # make payment
+        self.env['account.payment'].create({
+            'payment_type': 'inbound',
+            'payment_method_id': self.env.ref('account.account_payment_method_manual_in').id,
+            'partner_type': 'customer',
+            'partner_id': self.partner_a.id,
+            'amount': 1100,
+            'payment_date': move.date,
+            'journal_id': self.company_data['default_journal_bank'].id,
+            'invoice_ids': [(6, 0, move.ids)],
+        }).post()
+        # check caba move
+        partial_rec = move.mapped('line_ids.matched_credit_ids')
+        caba_move = self.env['account.move'].search([('tax_cash_basis_rec_id', '=', partial_rec.id)])
+        expected_values = [
+            {
+                'tax_line_id': False,
+                'tax_repartition_line_id': False,
+                'tax_ids': [],
+                'tag_ids': [],
+                'account_id': tax_waiting_account.id,
+                'debit': 100.0,
+                'credit': 0.0,
+            },
+            {
+                'tax_line_id': tax.id,
+                'tax_repartition_line_id': tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'tax').id,
+                'tax_ids': [],
+                'tag_ids': tax_tags['invoice']['tax'].ids,
+                'account_id': tax_final_account.id,
+                'debit': 0.0,
+                'credit': 100.0,
+            },
+            {
+                'tax_line_id': False,
+                'tax_repartition_line_id': False,
+                'tax_ids': tax.ids,
+                'tag_ids': tax_tags['invoice']['base'].ids,
+                'account_id': tax_base_amount_account.id,
+                'debit': 0.0,
+                'credit': 1000.0,
+            },
+            {
+                'tax_line_id': False,
+                'tax_repartition_line_id': False,
+                'tax_ids': [],
+                'tag_ids': [],
+                'account_id': tax_base_amount_account.id,
+                'debit': 1000.0,
+                'credit': 0.0,
+            },
+        ]
+        self.assertRecordValues(caba_move.line_ids, expected_values)
+        # unreconcile
+        debit_aml = move.line_ids.filtered('debit')
+        debit_aml.with_context(move_id=move.id).remove_move_reconcile()
```

Comment:
I don't understand why you need this `with_context`

Reply:
removed